src/librbd/ImageWatcher.cc

```diff
@@ -896,16 +891,6 @@ void ImageWatcher::handle_payload(const SnapCreatePayload &payload,
     int r = librbd::snap_create(&m_image_ctx, payload.snap_name.c_str(), false);
 
     ::encode(ResponseMessage(r), *out);
-    if (r == 0) {
-      // increment now to avoid race due to the delayed notification
-      Mutex::Locker lictx(m_image_ctx.refresh_lock);
-      ++m_image_ctx.refresh_seq;
```

Comment:
looks good to remove this, add_snap() is updating the lock owner's in-memory state for the new snap already